rfcs/react-components/convergence/semantic-shared-colors.md

```diff
@@ -0,0 +1,195 @@
+
+> Do not use CSS variables directly
+>
+> ⚠ Never use theme CSS variables directly! The CSS variables implementation of the theme is internal to the library. We might eventually decide to change the variable names, hash them or even use direct values instead of some variables. Always use the tokens to access the theme.
+
+Add "status" slots to `cranberry`, add `orange` as a status color.
+
+Add semantic color tokens which are mapped to the current shared colors. For example, `colorPaletteDangerBackground1` is mapped to `--colorPaletteRedBackground1`. `Danger` is still mapped to the old `red` color, but the token name is now semantic. Still not changed to `cranberry`.
+
+Update the sources in the library to use the new colors.
+
+Update the mapping from old colors to the new colors.
+
+#### 4B: Add CSS variables for new status tokens, map old tokens to the status CSS variables
+
+The variant 4 proposes to keep `--colorPaletteRedBackground1`, create new token `colorPaletteDangerBackground1` and map it to the red CSS variable.
+We can also implement this the other way around - create `colorPaletteDangerBackground1` mapped 1:1 to `--colorPaletteDangerBackground1` and re-map the legacy tokens `colorPaletteRedBackground1` to the status CSS variables.
+
+#### Pros and Cons
+
+- 👍 Semantic tokens without any unnecessary tokens.
+- 👍 Controlled partner upgrade from old hex values to new values, but coupled with library upgrade.
+- 👎 Semantic and status tokens are still coupled - the only way for the application to override `danger` is to override `red` (which will affect other parts of the UI).
+- 👎 Debugging might be harder - the CSS variable names are not the same as the token names.
+- 👎 There might be code which depends on the 1:1 token to CSS variable mapping (hello, [`themeToTokensObject`](https://github.com/microsoft/fluentui/blob/97c1c1ab218afb20e37bfc35936fd88177d439b4/packages/tokens/src/themeToTokensObject.ts#L14-L16)).
+- 👍 4B: The css variable names map to the new, recommended token names for easier debugging.
+- 👎 4B: If anyone is using the css variable names directly, they will break when we rename them to the new names.
+- ‼️ In the current implementation we use `Theme` type for theme definition and `Record<keyof Theme, string>` for tokens in styles. In option 4 this would have to change - see [Open Issues](#open-issues).
+
+## Naming the status tokens
+
+In the theme we currently use prefixes to have "namespaces" for individual color ramps - we use `color` prefix for neutral color ramp and `colorPalette` prefix for shared colors.
+
+What do we want to use for status tokens?
+
+- K: `colorPalette` (`colorPaletteDangerBackground1`)
+- L: `colorStatus` (`colorStatusDangerBackground1`)
+- M: `colorSemantic` (`colorSemanticDangerBackground1`)
+- n: anything else?
+
+Also consider how the proposed name would work if we decide to introduce semantic **presence** tokens (`away`).
```

Comment:
I like `colorStatus`. If we add presence colors, we could use `colorStatus` for that as well, or `colorPresence` if we wanted to distinguish them.